pynestml/codegeneration/nest_builder.py

```diff
@@ -86,24 +86,44 @@ def __init__(self, options: Optional[Mapping[str, Any]] = None):
             try:
                 import nest
             except ModuleNotFoundError:
-                Logger.log_message(None, -1, "An error occurred while importing the `nest` module in Python. Please check your NEST installation-related environment variables and paths.", None, LoggingLevel.ERROR)
+                msg = "An error occurred while importing the 'nest' module in Python. Please check your NEST installation-related environment variables and paths."
+                Logger.log_message(None, -1, msg, None, LoggingLevel.ERROR)
                 sys.exit(1)
 
             nest_path = nest.ll_api.sli_func("statusdict/prefix ::")
             self.set_options({"nest_path": nest_path})
-            Logger.log_message(None, -1, "The NEST installation was automatically detected as: " + nest_path, None, LoggingLevel.INFO)
+            msg = "The NEST installation was automatically detected as: " + nest_path
+            Logger.log_message(None, -1, msg, None, LoggingLevel.INFO)
 
-    def build(self) -> None:
+    def build(self, output_file: TextIO = None, error_file: TextIO = None) -> None:
         r"""
         This method can be used to build the generated code and install the resulting extension module into NEST.
 
+        Parameters
+        ----------
+        output_file
+            A file object containing the summary of the 'cmake' executed command.
+            By default the output_file will be set to the stderr output
+
+        error_file
+            A file object containing the errors thrown by the 'cmake' executed command.
+            By default the error_file will be set to the stderr output
+
+
         Raises
         ------
         GeneratedCodeBuildException
             If any kind of failure occurs during cmake configuration, build, or install.
         InvalidPathException
             If a failure occurs while trying to access the target path or the NEST installation path.
         """
+        error_location = ""
+        if error_file is None:
+            error_file = subprocess.STDOUT
```

Comment:
Any particular reason for redirecting stderr to stdout by default?

Reply:
If you take a look at the original code before introducing this modification in the `subprocess.check_call` function call, you will see that the ´stderr´ argument in the function was set to the `STDOUT`. Since I am also printing everything to a file, In line 121, I am just checking either to print everything in the STDOUT or in the file. Correct me if wrong.